File: codegen/results/v002/ollama/qwen3.5_35b-a3b-coding-nvfp4_nothink/20260414_100330/go/005_unit_test_writer/solution.go
package main

import (
	"fmt"
	"math"
)

func calculateDiscount(price float64, quantity int) float64 {
	if quantity < 10 {
		return price * float64(quantity)
	} else if quantity < 50 {
		return price * float64(quantity) * 0.9
	}
	return price * float64(quantity) * 0.75
}

func main() {
	// Helper to compare floats with a small tolerance
	equal := func(a, b float64) bool {
		diff := math.Abs(a - b)
		return diff < 1e-6
	}

	runTest := func(desc string, price float64, qty int, expected float64) {
		got := calculateDiscount(price, float64(qty))
		if !equal(got, expected) {
			fmt.Printf("FAIL: %s (expected: %v, got: %v)\n", desc, expected, got)
		} else {
			fmt.Printf("PASS: %s\n", desc)
		}
	}

	// Test cases covering boundary conditions:
	// quantity < 10 (boundary at 9)
	// 10 <= quantity < 50 (boundary at 10, 49)
	// quantity >= 50 (boundary at 50)

	// Test 1: quantity = 9 (just below first discount threshold)
	runTest("quantity=9 (no discount)", 10.0, 9, 90.0)

	// Test 2: quantity = 10 (first discount threshold: 10% off)
	runTest("quantity=10 (10% discount)", 10.0, 10, 10.0*10*0.9) // 90.0

	// Test 3: quantity = 49 (still in 10% discount range)
	runTest("quantity=49 (10% discount)", 10.0, 49, 10.0*49*0.9) // 441.0

	// Test 4: quantity = 50 (second discount threshold: 25% off)
	runTest("quantity=50 (25% discount)", 10.0, 50, 10.0*50*0.75) // 375.0
}

File: codegen/results/v002/ollama/qwen3.5_35b-a3b-coding-nvfp4_nothink/20260414_100330/python/005_unit_test_writer/solution.py
def calculate_discount(price: float, quantity: int) -> float:
    """
    Calculate total price after bulk discount.
    quantity < 10: no discount
    quantity 10-49: 10% off
    quantity >= 50: 25% off
    Returns: price * quantity * discount_multiplier
    """
    if quantity < 10:
        return price * quantity
    elif quantity < 50:
        return price * quantity * 0.9
    else:
        return price * quantity * 0.75


def run_tests():
    test_cases = [
        # (price, quantity, expected_result, description)
        (100.0, 9, 900.0, "quantity < 10 (no discount)"),
        (100.0, 10, 900.0, "quantity = 10 (10% discount boundary)"),
        (100.0, 49, 4410.0, "quantity = 49 (10% discount boundary)"),
        (100.0, 50, 3750.0, "quantity = 50 (25% discount boundary)"),
        (50.0, 5, 250.0, "quantity < 10 (no discount, low price)"),
        (200.0, 15, 2700.0, "quantity in 10-49 range"),
        (200.0, 60, 9000.0, "quantity >= 50 range"),
    ]

    for price, quantity, expected, description in test_cases:
        result = calculate_discount(price, quantity)
        # Use a small tolerance for floating point comparison
        if abs(result - expected) < 1e-6:
            print(f"PASS: {description}")
        else:
            print(f"FAIL: {description} (expected: {expected}, got: {result})")


if __name__ == "__main__":
    run_tests()